### ISSUE TITLE
jsconfig.json generates incorrectly and disallows correction

### ISSUE THREAD

**Issue by djeune** [forcedotcom/salesforcedx-vscode#5520]
### Summary

When the Salesforce Extensions for VSCode activate, they generate a jsconfig.json file within the lwc directory to support intellisense in Lightning Web Components.

When the file is generated, it includes a key "paths" as a top level property of the configuration object. This is the incorrect place for the key, as it belongs within the compilerOptions object.

See [documentation here](https://code.visualstudio.com/docs/languages/jsconfig) for more information. Specifically, the "jsConfig options" header describes the correct configuration.

To make the issue worse, this improper configuration is re-applied after removal any time the Salesforce Extensions reload.

### Steps To Reproduce:

1.  Generate a new project with the Salesforce CLI command `sf project generate`
1.  Observe the generated jsconfig.json file within the lwc directory and its improper structure.
1.  Move the key to the compilerOptions object and save the file.
1.  Reload the window from the command palette to force the Salesforce Extensions to reload.
1.  Observe that the erroneous key has been re-added to the configuration.

### Expected result

The jsconfig.json file should generate with a valid configuration file. Failing that, it should respect the user's attempts to fix it.

### Actual result

The jsconfig.json file generates with an invalid "paths" key and re-adds this key when it is removed.

### Additional information

**Salesforce Extension Version in VS Code**: 60.5.1

**SFDX CLI Version**: 2.33.3

**OS and version**: MacOS Sonoma 14.3.1

**VS Code version**: 1.87.2

Additional note on preferred fix: I would prefer to be able to modify files after generation without worrying about the extensions overriding my changes. I have also run into this problem with .forceignore, which doesn't allow me to the remove .eslint.json entry even though I am using the flat config eslint.config.js file located at the root of the project.

**Comment by CristiCanizales** [comment]
Hi @djeune, thank you for bringing this to out attention. I was able to replicate this bug and we'll be adding it to our backlog.

**Comment by CristiCanizales** [comment]
Link to Bug [W-15374337](https://gus.my.salesforce.com/a07EE00001nPNRjYAO)

**Comment by diyer** [comment]
@djeune can you please describe the user impact on development due to this behavior?

**Comment by djeune** [comment]
The configuration file is necessary for type acquisition when attempting to use VSCode's built-in javascript/typescript features.

When I discovered the problem, I was attempting to get type information working from one LWC component to another. For example, if I had an LWC component that imported shared code from a different component, I wanted vscode to support type information from that shared code. [Process described here](https://developer.salesforce.com/docs/platform/lwc/guide/js-share-code.html).

I reported the erroneous file generation because is generating invalid configuration files. However, the best case would be if the standard configuration supported the use case linked above.
